Emil Medve <Emilian.Medve@Freescale.com>

Somewhere between verson 1.3 and 1.4.1 of genext2fs the semantic of the -i command line option
changed. In 1.4.1 -N plays the same functinal role -i used to play in 1.3. Due to backwards
compatibility concerns this patch ensures that when running from LTIB, the -i command line option
has the same semantic as in 1.3

diff -bBwd -uNr genext2fs-1.4.1.orig/genext2fs.c genext2fs-1.4.1/genext2fs.c
--- genext2fs-1.4.1.orig/genext2fs.c	2007-03-26 09:19:59.000000000 -0500
+++ genext2fs-1.4.1/genext2fs.c	2008-02-06 08:39:12.000000000 -0600
@@ -2514,7 +2514,15 @@
 				nbblocks = SI_atof(optarg);
 				break;
 			case 'i':
+				if(getenv("SPOOF_PATH") != NULL)
+				{
+					fprintf(stderr, "%s: Running in LTIB backwards compatibility mode: -i -> -N\n", app_name);
+					nbinodes = SI_atof(optarg);
+				}
+				else
+				{
 				bytes_per_inode = SI_atof(optarg);
+				}
 				break;
 			case 'N':
 				nbinodes = SI_atof(optarg);
